### ISSUE TITLE
flags not defined

### ISSUE THREAD

**Issue by BenOlley** [fabriceb/gcalcron#15]
root@RUBY:~/gcalcron# python gcalcron.py
ERROR:root:Sync failed
Traceback (most recent call last):
  File "gcalcron.py", line 427, in main
    g.sync_gcal_to_cron()
  File "gcalcron.py", line 300, in sync_gcal_to_cron
    events = self.gCalAdapter.get_events(sync_start, last_sync, num_days)
  File "gcalcron.py", line 170, in get_events
    return self.queryApi(queries)
  File "gcalcron.py", line 137, in queryApi
    gCalEvents = self.get_service().events().list(**query).execute()
  File "gcalcron.py", line 84, in get_service
    credentials = tools.run_flow(FLOW, storage, Flags)
NameError: global name 'flags' is not defined
root@RUBY:~/gcalcron#

I have had this issue as-well as the initial last_sync being null and throwing out allot of errors, I placed yesterdays date in for a initial run and received this

**Comment by fabriceb** [comment]
What is your python version?

**Comment by rplink3r** [comment]
Hello, I have the same problem, my version of python is 2.7.3 running on raspberry pi, I have all prerequisites installed, checked the .json file to inclue my email.

**Comment by clederman** [comment]
Hello, 

I have the same problem, "NameError: global name 'flags' is not defined", I'm also on raspberry PI with python 2.7.3. 
I solved the problem by adding the line 406 -> global flags. 

Against by now I have a error message "Bad request".

**Comment by mattman73** [comment]
Hi 
Fixed the problem by adding global flags on the line under # Parse the command-line flags.
Regards
Matt

**Comment by coacheli** [comment]
I am having the same problem as the original poster.  Can you explain your solution to a relative newbie?

**Comment by coacheli** [comment]
OK, if I'm reading correctly, you added "  global flags" to the line under "#Parse the command-line flags".  I did it and this did fix the issue.  However now it keeps asking to authorize the account but then it opens a bad google link.  It's trying to get a verification code but I think something is wrong with the URL

**Comment by Kayto** [comment]
Same issues as above. Seems an issue with the Google API consent screen. I cannot see a way to bypass this. Anyone had any success?

**Comment by Kayto** [comment]
Think I have resolved. 

From Command line.
python gcalcron.py --noauth_local_webserver

You will be presented with a internet address and requirement for verification code. Copy web link into browser and Accept. Google will provide a verification code. Copy out from web browser and into command line request.

Done.

**Comment by coacheli** [comment]
I figured out the verification code thing.  First, I had to go to my API setup ( https://console.developers.google.com/project ) and give my application a name, as found here: http://stackoverflow.com/questions/18677244/error-invalid-client-no-application-name

Then I ran "python gcalcron.py" from the Raspberry Pi desktop terminal (not putty).  That made the link pop up in the Pi web browser.  If I was logged in, it would declare "the authentication flow has completed"

Sounds good, right?  But then I get a new list of errors:  

root@raspberrypi:/home/pi/gcalcron# python gcalcron.py
ERROR:root:Sync failed
Traceback (most recent call last):
  File "gcalcron.py", line 427, in main
    g.sync_gcal_to_cron()
  File "gcalcron.py", line 306, in sync_gcal_to_cron
    self.schedule_new_jobs(commandsList)
  File "gcalcron.py", line 262, in schedule_new_jobs
    p = subprocess.Popen(cmd, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
  File "/usr/lib/python2.7/subprocess.py", line 679, in **init**
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1259, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

It seems to be looking for a file but not finding it.
